tt_metal/impl/device/device.hpp

```diff
@@ -193,13 +210,25 @@ class Device {
     void compile_command_queue_programs_for_grayskull();
     void configure_command_queue_programs();
     void clear_l1_state();
-
     std::pair<int, int> build_processor_type_to_index(JitBuildProcessorType t) const;
 
     // Puts device into reset
     bool close();
     friend bool CloseDevice(Device *device);
 
+    void start_worker();
+    void run_worker();
+    void stop_worker();
+    void push_work(std::function<void()> work_executor, bool blocking = false);
+    void synchronize();
+    void set_worker_mode(const WorkerQueueMode& mode);
+    void enable_async(bool enable);
+    static WorkerQueueMode get_worker_mode() { return worker_queue_mode; }
+    static WorkerQueueMode default_worker_queue_mode() {
+        static int value = parse_env<int>("TT_METAL_ASYNC_DEVICE_QUEUE", static_cast<int>(WorkerQueueMode::SYNCHRONOUS));
```

Comment:
The merge will continue doing synchronous dispatch?

Reply:
Synchronous dispatch is default, since not all ops support async. Async dispatch is opt in (running on some CI tests).

Reply:
sounds good!